wordpress-mobile/WordPress-Android
# WP Stories integration - Media Picker part 2: using EditorMedia to source from the picker

This PR builds on top of #11939

This PR implements the changes needed for StoryComposerActivity to bridge its interactions with the MediaPicker through EditorMedia, in the same way the EditPostActivity works.

This PR introduces a change in EditorMedia to be able to tell its use cases whether upload is optional or necessary. Up til now, we were enqueueing images added to a Post immediately for upload, but with Stories it's a little bit different, as we only want to upload the composed images (images plus text and emoji) in the Post. The EditorMedia encapsulates the different problems from taking media from different sources and making them available to an Editor very well so, it makes sense to follow the same paths and only diverge in the upload part.

We introduced then, a flag called
```
enum class EditorType {
    POST_EDITOR,
    STORY_EDITOR
}
```
- If EditorMedia is being used in the context of an editor of type `POST_EDITOR`, then its flow will include uploading, as normally.
- If EditorMedia is being used in the context of an editor of type `STORY_EDITOR`, then upload is not made part of the flow, as simple as that.

We will split the upload part later, once the Story frames have been locally saved and are available to be actually uploaded to the backend.

Note for now this only has the media from the device path covered (to keep PRs simple). We'll work on WPMedia and Free Photo source paths on subsequent PRs.


To test:
1. run the app
2. select a site
3. tap the FAB button on the "MySite" screen
4. select Story
5. observe the Story creation experience appears (camera preview)
6. tap on the upload icon to the left of the camera shutter
7. observe the WPAndroid media picker gets launched
8. select some media items and tap on the checkmark on the top toolbar to indicate you're done selecting
9. observe these media items are added as background for as many Story pages as media items were selected.

Also test:

0. fresh install
1. repeat steps 1-8 above
2. the image optimisation dialog should appear. Verify it appears correctly and the app doesn't crash (this was saved by applying the correct style in 12ca7d3)

PR submission checklist:
